File: 006.rb
class SumSquareDifference
    attr_reader :answer

    def initialize(range)
        @range = range
        @answer = difference
    end

    def difference
        sum_of_squares - square_of_sum
    end

    def sum_of_squares
        @range.map{|n| n**2}.reduce(:+)
    end

    def square_of_sum
        (@range.reduce(:+))**2
    end
end

puts SumSquareDifference.new(1..10).answer

File: 006.js
'use strict';

var square = function(n) {
    return n*n;
};

var difference = function(n, m) {
    return n - m;
};

var sumOfSquares = function(min, max) {
    var range = [];
    var sum;
    for (var i = min; i <= max; i++) {
        range.push(square(i));
    }
    return range.reduce(function(a, b)     {
        return a+b;
    });
};

var squareOfSum = function(min, max) {
    var range = [];
    var sum;
    for (var i = min; i <= max; i++) {
        range.push(i);
    }
    sum = range.reduce(function(a, b)     {
        return a+b;
    });
    return Math.pow(sum, 2);
};

difference(sumOfSquares(1, 100), squareOfSum(1, 100));
